ddtrace/llmobs/_integrations/langchain.py

```diff
@@ -20,6 +28,114 @@
 class LangChainIntegration(BaseLLMIntegration):
     _integration_name = "langchain"
 
+    def llmobs_set_tags(
+        self,
+        operation: str,  # oneof "llm","chat_model","chain"
+        span: Span,
+        inputs: Any,
+        response: Any,
+        error: bool = False,
+    ) -> None:
+        """Sets meta tags and metrics for span events to be sent to LLMObs."""
+        if not self.llmobs_enabled:
+            return
+        model_provider = span.get_tag(PROVIDER)
+        span.set_tag_str(SPAN_KIND, "llm")
+        span.set_tag_str(MODEL_NAME, span.get_tag(MODEL) or "")
+        span.set_tag_str(MODEL_PROVIDER, model_provider or "")
+
+        # input parameters
+        if model_provider:
+            input_parameters = {}
+            temperature = span.get_tag(f"langchain.request.{model_provider}.parameters.temperature")
+            max_tokens = span.get_tag(f"langchain.request.{model_provider}.parameters.max_tokens")
+            if temperature:
+                input_parameters["temperature"] = float(temperature)
+            if max_tokens:
+                input_parameters["max_tokens"] = int(max_tokens)
+            if input_parameters:
+                span.set_tag_str(INPUT_PARAMETERS, json.dumps(input_parameters))
+
+        if operation == "llm":
+            self._llmobs_set_meta_tags_from_llm(span, inputs, response, error)
+        elif operation == "chat_model":
+            self._llmobs_set_meta_tags_from_chat_model(span, inputs, response, error)
+        elif operation == "chain":
+            # TODO to be added as a follow-up PR
+            pass
+        self._llmobs_set_metrics_tags(span)
+
+    def _llmobs_set_meta_tags_from_llm(
+        self,
+        span: Span,
+        prompts: List[Any],
+        completions: Any,
+        err: bool = False,
+    ) -> None:
+        # input messages
+        if isinstance(prompts, str):
+            prompts = [prompts]
+        span.set_tag_str(INPUT_MESSAGES, json.dumps([{"content": str(prompt)} for prompt in prompts]))
+
+        # output messages
+        message_content = [{"content": ""}]
+        if not err:
+            message_content = [{"content": self.trunc(completion[0].text)} for completion in completions.generations]
```

Comment:
For regular APM langchain/openai integrations, truncation is required for prompt/responses for storage/privacy limits, but for LLMObs we need the entire prompt/response so let's remove the truncation here

Also, need to double check why we do `completion[0].text` instead of `completion` (for my own sanity)

Reply:
ah yeah that totally makes sense, will change here (and the other places).

and yeah, will do!

Reply:
so it looks like completions is of type `List[List[Generation]]`, even on older versions of LangChain (debugging `langchain==0.0.192`, the older version that we test with, it is also that type). I copypasta'd this from `patch`, but I think we do it this way since it's a nested list of `Generation` objects, so we have to access the `.text` field. I suppose we take the first one for simplicity in matching 1:1 for prompts/responses for LLMs, but correct me if I'm wrong!